drivers/flash/flash_sam0.c

```diff
@@ -67,7 +70,7 @@ static const struct flash_parameters flash_sam0_parameters = {
 #if CONFIG_SOC_FLASH_SAM0_EMULATE_BYTE_PAGES
 	.write_block_size = 1,
 #else
-	.write_block_size = DT_PROP(DT_INST(0, soc_nv_flash), write_block_size),
+	.write_block_size = FLASH_WRITE_BLK_SZ,
```

Comment:
You could add `BUILD_ASSERT((FLASH_WRITE_BLK_SZ % sizeof(uint32_t)) == 0, "unsupported write-block-size");`

Reply:
Added it right before flash_sam0_write...